### ISSUE TITLE
Crash when inserting a record with a DATETIME with CURRENT_TIMESTAMP as default

### ISSUE THREAD

**Issue by jcherniak** [yiisoft/yii#2993]
MySQL 5.6 supports having DATETIME columns with TIMESTAMP semantics.  Among other things, they support having DATETIME columns with CURRENT_TIMESTAMP default values.

If you try and save a CActiveRecord which has one of these columns, you see the following error:
    CDbException - CDbCommand failed to execute the SQL statement: SQLSTATE[22007]: Invalid datetime format: 1292 Incorrect datetime value: 'CURRENT_TIMESTAMP' for column 'last_updated' at row 1.

It seems that the framework is interpreting the CURRENT_TIMESTAMP value as a string instead of a CDbExpression

**Comment by tomvdp** [comment]
FWIW, I just hit exactly this bug.
I need a datetime with default CURRENT_TIMESTAMP.
Why does Yii try to supply values when the database has already defined its own defaults ?

(still working in Yii 1.1 - might be resolved in Yii2?)

**Comment by samdark** [comment]
@tomvdp do you have time for a pull request?

**Comment by tomvdp** [comment]
Done!

**Comment by machei** [comment]
Fix for maria db latest version 10.2.9. 


```
if(strncmp($this->dbType,'bit',3)===0)
			$this->defaultValue=bindec(trim($defaultValue,'b\''));
		elseif(($this->dbType==='timestamp' || $this->dbType==='datetime')  && ($defaultValue==='CURRENT_TIMESTAMP' || $defaultValue==='current_timestamp()') ){
			$this->defaultValue=null;
		}else
			parent::extractDefault($defaultValue);
```
